test/e2e/e2e_test.go

```diff
@@ -186,33 +186,33 @@ func (t *PorchSuite) TestCloneFromUpstream(ctx context.Context) {
 	// Register Upstream Repository
 	t.registerGitRepositoryF(ctx, testBlueprintsRepo, "test-blueprints", "")
 
-	var list porchapi.PackageRevisionList
+	var list porchapi.PorchPkgRevisionList
 	t.ListE(ctx, &list, client.InNamespace(t.namespace))
 
-	basens := MustFindPackageRevision(t.T, &list, repository.PackageRevisionKey{Repository: "test-blueprints", Package: "basens", Revision: "v1"})
+	basens := MustFindPorchPkgRevision(t.T, &list, repository.PackageRevisionKey{Repository: "test-blueprints", Package: "basens", Revision: "v1"})
 
 	// Register the repository as 'downstream'
 	t.registerMainGitRepositoryF(ctx, "downstream")
 
-	// Create PackageRevision from upstream repo
-	pr := &porchapi.PackageRevision{
+	// Create PorchPkgRevision from upstream repo
+	pr := &porchapi.PorchPkgRevision{
 		TypeMeta: metav1.TypeMeta{
-			Kind:       "PackageRevision",
+			Kind:       "PorchPkgRevision",
```

Comment:
The source of my worry is with these test cases, where you need to change the TypeMeta.kind, which will be serialized as-is. And if this is accepted, this means the accepted kind is changing as well.

Reply:
I would tend to agree. The main problem is with the Package CR. All other CRs should remain as is.
I have been hacking at this for a while now. PR pushed here but still work in progress.
https://github.com/nephio-project/porch/pull/35
The dependency upgrades have broken quite a bit. I have fixed most of it but still some issue with the api.
This change in particular needs to be remedied I believe.
https://kubernetes.io/blog/2023/04/24/openapi-v3-field-validation-ga/
At present, the api server impl doesn't have any openapi config.